documentation/specs/proposed/BuildCheck.md

```diff
@@ -0,0 +1,405 @@
+* Analyzers reports (errors, warnings, messages) are in logger output, VS error window.
+* Codes will be distinguishable from standard build warnings/error (to prevent easy mixups and attempts to configure standard msbuild warnings/errors via editorconfig), but otherwise the outputs are very similar.
+* Default opt-ins and levels for inbox analyzers set by sdk version (via [`$SdkAnalysisLevel`]((https://github.com/dotnet/designs/blob/main/proposed/sdk-analysis-level.md))) or other agreed mechanism for controling increasing strictness between .NET versions.
+* Custom analyzers opted in via `PackageReference` of a particular nuget with the analyzer.
+* Explicit overrides of enablement and analysis levels via `.editorconfig` file (with up to a per project scope).
+* Standards of `.editorconfig`s will be observed. 
+* Simplified authoring experience via template and doc.
+* Single analyzer can produce reports for multiple rules. However those need to be declared upfront.
+* Opt-in reporting of time spent via specific analyzers and infra overall.
+* Collect touched `.editorconfg`s into binlog embedded files.
+* Possibility to opt-out from analysis - the perf should not be impacted when done so.
+* Team collects performance impact numbers on a set of benchmark builds with the inbox analyzers enabled.
+
+**Non Goals, but subject for consideration**
+* Custom anlyzer in a local project (source codes) or a binary.
+* Bulk configuration of multiple rules (based on prefixes).
+* Specifying scope of MSBuild imports that will be considered for analysis (so that e.g. data from sdk won't even be passed to anlyzer, if not requested).
+* Attempts to try to configure standard msbuild warnings/errors via `.editorconfig` should lead to fail fast errors.
+* Configuring analysis levels when analysing from binlog - beyond the collected editorconfigs
+* Rich information in VS error window.
+
+
+**Out of scope**
+* Custom analyzers has equal data access as the inbox analyzers.
+* All inbox analyzers reports have precise location(s) of issues.
+* Opt-out of analysis on code-level (analogy to C# pragmas, but within msbuild xml files).
+* Simplified authoring experience via dedicated reference assembly.
+* Restore phase analysis.
+* Turning analysis off/on based on target (e.g. multi-targetted builds, calling MSBuild task etc.).
+* Controling/Configuring lifetime of analyzers - analyzers will currently be held alive, as single instance per analyzer, for the whole duration of the build. But future versions might prevent some of the analyzers to survive beyond a scope of a single project built (means for sharing data would be provided).
+* ETW for analyzers.
+* Attributing `.editorconfig` configurations to .sln files.
+* Attributing `.editorconfig` configurations to lower granularity than whole projects.
+* Respecting `.editorconfig` file in msbuild import locations (unless they are in the parent folders hieararchy of particular project file).
+ 
+
+# User Experience
```

Comment:
I'd move this section above the initial scope stuff I think.

Reply:
I prefer to have the scope first now - as that's where I want most attention of initial readers (to get early feedback if something important is not properly considered).

We can rearange once this becomes more of a user doc.